esphome/components/esp32_touch/binary_sensor.py

```diff
@@ -52,9 +78,10 @@ def validate_touch_pad(value):
 
 async def to_code(config):
     hub = await cg.get_variable(config[CONF_ESP32_TOUCH_ID])
+    variant = get_esp32_variant()
     var = cg.new_Pvariable(
         config[CONF_ID],
-        TOUCH_PADS[config[CONF_PIN]],
+        TOUCH_PADS[variant][config[CONF_PIN]],
```

Comment:
As per above, you would get the `cg.global_ns.TOUCH_PAD_NUMx` value here directly like this.
```suggestion
        config[CONF_PIN],
```

Reply:
Thanks for the feedback! I got a bit sidetracked with other stuff but will fix this soon (if it's not already fixed in the main branch by now)